[Cedric Chia] iP

CedricChia123 opened the pull request:
### Qoobee

> "Deciding what NOT to do is as important as deciding what to do." - Steve Jobs [(source)](https://www.linkedin.com/pulse/deciding-what-do-important-steve-jobs-alex-tsouloftas#:~:text=%E2%80%9CDeciding%20what%20not%20to%20do,what%20to%20do%E2%80%9D%20%2D%20Steve%20Jobs)

Qoobee helps keep track of your tasks with simple functionalities, making sure your deadlines are met. It's,

- text-based
- **VERY** simple to use
- a friendly bot ^.^

All you need to do is,

1. download it from [here](https://github.com/CedricChia123/ip)
2. double-click it
3. type simple commands to add tasks
4. let it organise your tasks neatly 🤩

And that is all!

Features: 

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use Qoobee to practice your coding too. Here is the main method:

```
public static void main(String[] args) {
        Qoobee qoobee = new Qoobee();
        qoobee.run();
}
```

Commit:
Add Gradle support

CedricChia123 committed:
added echo commands

CedricChia123 committed:
allow use of list

CedricChia123 committed:
Implemented Task class, completed mark and unmark methods

CedricChia123 committed:
Added deadline, events

CedricChia123 committed:
Added deadline, events

CedricChia123 committed:
changed formatting, added test cases

CedricChia123 committed:
handled exceptions when user inputs invalid command

CedricChia123 committed:
added delete function

CedricChia123 committed:
changed string formatting

CedricChia123 committed:
Add saving to hard disk

(cherry picked from commit e4ea9d20bb31128c5fe2ab78014b2c8fd78591a0)

CedricChia123 committed:
Add saving to hard disk

CedricChia123 committed:
Revert "Add saving to hard disk"

This reverts commit e4ea9d20bb31128c5fe2ab78014b2c8fd78591a0.

CedricChia123 committed:
Add localdatetime for Deadline

CedricChia123 committed:
Add OOP files Ui, Parser, TaskList, Storage

CedricChia123 committed:
Add packages

CedricChia123 committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

CedricChia123 committed:
Add gradle support

CedricChia123 committed:
Add Junit test cases

CedricChia123 committed:
Add jar file

CedricChia123 committed:
Add javadoc comments

CedricChia123 committed:
Change code to fit module coding standard

CedricChia123 committed:
Add command to find tasks in list

CedricChia123 committed:
Add javadoc comment for findtask

LianGuoYang commented on src/main/java/qoobee/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+    protected LocalDateTime dateTime;
+
+    /**
+     * Creates a deadline given a description and date.
```
Perhaps a line break between line 14 and 15? I also noticed similar issues at several other places.

LianGuoYang commented on src/test/java/qoobee/ToDoTest.java:
```diff
@@ -0,0 +1,14 @@
+    public void toStringTest() {
+        assertEquals("[T][ ] do CS2103T iP", new ToDo("do CS2103T iP").toString());
+    }
+
```
Perhaps removing extra line break here?

LianGuoYang commented on src/main/java/qoobee/Ui.java:
```diff
@@ -0,0 +1,47 @@
+    public void greet() {
+        System.out.println("Hello I'm best bot Qoobee!\n" + "What can I do for you? ^.^");
+    }
+
```
Perhaps removing extra line break here? I also noticed the same issues at several places.

LianGuoYang commented on src/main/java/qoobee/Storage.java:
```diff
@@ -0,0 +1,129 @@
+                    String[] time = dateTime[3].split(":", 2);
+                    Task deadline = new Deadline(stringArray[2],
+                            LocalDateTime.of(Integer.parseInt(dateTime[2]), Integer.parseInt(dateTime[1])
+                                    , Integer.parseInt(dateTime[0]), Integer.parseInt(time[0]),
```
Perhaps the comma should be at the previous line?

LianGuoYang commented on src/main/java/qoobee/Parser.java:
```diff
@@ -0,0 +1,95 @@
+                }
+                else {
+                    throw new QoobeeException("I'm sorry, but I don't know what that means :^(");
+                }
```
I like where you add multiple exceptions to account for different errors.

LianGuoYang commented on src/main/java/qoobee/Parser.java:
```diff
@@ -0,0 +1,95 @@
+            System.out.println(e);
+        }
+    }
+
```
Perhaps removing extra line break here?

LianGuoYang reviewed (commented):
LGTM. Minor mistakes but overall well done. I like how all of your variables are clearly named.

CedricChia123 committed:
Edit checkstyle

CedricChia123 committed:
Add javafx

Ugholaf commented on src/main/java/qoobee/Parser.java:
```diff
@@ -0,0 +1,95 @@
+                    return tasks.findTask(command[1]);
+                } else {
+                    return "I'm sorry, but I don't know what that means :^(";
+                }
```
Would it be better to use switch case here instead of multiple if statement?

Ugholaf commented on src/main/java/qoobee/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+    public Task getTask(int index) {
+        return this.taskList.get(index);
+    }
+
```
I like your naming of methods. The names are very descriptive and clearly convey what the methods are doing.

Ugholaf commented on src/main/java/qoobee/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+        }
+    }
+
+    /**
+     * Adds a task into the list.
+     * @param task A task to be added in the list.
+     * @throws QoobeeException if the user inputs an invalid syntax.
+     */
+    public String addTask(Task task) throws QoobeeException {
+        this.taskList.add(task);
+        storage.save(taskList);
+        return "Got it. I've added this task:\n" + task + "\n"
+                + "Now you have " + taskListSize() + " tasks in the list.";
+    }
```
```suggestion
    /**
     * Adds a task into the list.
     *
     * @param task A task to be added in the list.
     * @return A String with task added successfully message
     * @throws QoobeeException if the user inputs an invalid syntax.
     */
    public String addTask(Task task) throws QoobeeException {
        this.taskList.add(task);
        storage.save(taskList);
        return "Got it. I've added this task:\n" + task + "\n"
                + "Now you have " + taskListSize() + " tasks in the list.";
    }
```

I realised for most of the Javadoc comments, you have missed out on the return tag for methods with a return statement. Also, just a minor error, remember to leave an empty line between the description and the tags.

Ugholaf reviewed (commented):
Overall I really like your naming of methods and variables as it is very obvious what the methods and variables are for. However, there are some minor errors with the Javadoc comments, and if possible, try to use `switch case` if there are multiple `else if` statements as the code will look neater.

CedricChia123 committed:
Update java assertions

The assert feature serves to document important assumptions that should hold true at some points of the code.

CedricChia123 committed:
Update java assertions

The assert feature serves to document important assumptions that should hold true at some points of the code.

CedricChia123 committed:
Revert "Update java assertions"

This reverts commit f65a9dfe574a612836a77da64bd362d7381a190e.

CedricChia123 committed:
Fix coding standard.

There were several methods that were duplicated. Unifying the code improves the code quality.

Added a new datetimeparser to reduce the length of code to parse a string to a datetime.

CedricChia123 committed:
Merge pull request #2 from CedricChia123/branch-A-Assertions

Add some assertions to the code

CedricChia123 committed:
Merge pull request #3 from CedricChia123/branch-A-CodeQuality

Fix coding standard.

CedricChia123 committed:
Add extension C-Priority

CedricChia123 committed:
Add user guide

Qoobee user guide is added. Fixed checkstyle issues as well.

CedricChia123 committed:
Edit README

CedricChia123 committed:
Fix ReadMe issues

CedricChia123 committed:
Fix checkstyle issues

CedricChia123 committed:
Edit User Guide

CedricChia123 committed:
Update User Guide

Fix issues with GFMD
